src/contexts/AuthContextProvider.jsx

```diff
@@ -1,16 +1,50 @@
-import React, { createContext, useContext, useState } from 'react';
+import React, { createContext, useContext, useState, useEffect } from 'react';
 import { getAuth, onAuthStateChanged } from 'firebase/auth';
 
+import { doc, getDoc, setDoc } from 'firebase/firestore';
+import { db } from '../utils/FirebaseConfig';
+
 const authContext = createContext();
 
 export const useAuth = () => {
 	return useContext(authContext);
 };
 
 const AuthContextProvider = ({ children }) => {
+	const auth = getAuth();
+
 	const [user, setUser] = useState(undefined);
 
-	const auth = getAuth();
+	useEffect(() => {
+		const fetchData = async () => {
+			const docRef = doc(db, 'users', user.uid);
+			const docSnap = await getDoc(docRef);
+
+			if (docSnap.exists()) {
+				const data = docSnap.data();
+				if (data.extensionStates) {
+					// Get the init state for webview
+					window.api.send('extensionStates', data.extensionStates);
+				}
+			}
+		};
+
+		window.api.recieve('getExtensionStatesReply', () => {
+			if (user) fetchData();
+		});
+
+		window.api.recieve('setLatestOverlayLocationReply', (...payload) => {
+			if (user) {
+				const docRef = doc(db, 'users', user.uid);
+				setDoc(
+					docRef,
+					{ extensionStates: { scrollHelpPosition: { top: payload[0][0], left: payload[0][1] } } },
+					{ merge: true },
+				);
+			}
+		});
+		if (user) fetchData();
```

Comment:
Wat ik hier meestal doe is een `if (!user) return` als eerste regel. Dan moet je dat niet meerdere keren checken ;) En worden die functies niet aangemaakt als de user undefined is